backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/payment/C_PaySelection_StepDef.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * #%L
+ * de.metas.cucumber
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.cucumber.stepdefs.payment;
+
+import de.metas.banking.PaySelectionId;
+import de.metas.banking.payment.IPaySelectionBL;
+import de.metas.banking.process.C_PaySelection_CreateFrom;
+import de.metas.banking.process.C_PaySelection_CreatePayments;
+import de.metas.bpartner.BPartnerId;
+import de.metas.cucumber.stepdefs.C_BP_BankAccount_StepDefData;
+import de.metas.cucumber.stepdefs.C_BPartner_StepDefData;
+import de.metas.cucumber.stepdefs.DataTableRow;
+import de.metas.cucumber.stepdefs.DataTableRows;
+import de.metas.cucumber.stepdefs.ValueAndName;
+import de.metas.document.engine.IDocument;
+import de.metas.document.engine.IDocumentBL;
+import de.metas.payment.PaymentId;
+import de.metas.process.AdProcessId;
+import de.metas.process.IADProcessDAO;
+import de.metas.process.ProcessInfo;
+import de.metas.util.Services;
+import io.cucumber.datatable.DataTable;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Then;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.adempiere.exceptions.AdempiereException;
+import org.compiere.model.I_C_PaySelection;
+import org.compiere.model.I_C_PaySelectionLine;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.adempiere.model.InterfaceWrapperHelper.newInstance;
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@RequiredArgsConstructor
+public class C_PaySelection_StepDef
+{
+	private final @NonNull C_PaySelection_StepDefData paySelectionTable;
+	private final @NonNull C_PaySelectionLine_StepDefData paySelectionLineTable;
+	private final @NonNull C_BP_BankAccount_StepDefData bankAccountTable;
+	private final @NonNull C_Payment_StepDefData paymentTable;
+	private final @NonNull C_BPartner_StepDefData bpartnerTable;
+	private final @NonNull IDocumentBL documentBL = Services.get(IDocumentBL.class);
+	private final @NonNull IPaySelectionBL paySelectionBL = Services.get(IPaySelectionBL.class);
+
+	private final IADProcessDAO adProcessDAO = Services.get(IADProcessDAO.class);
```

Comment:
pedantry:
* why this service has a space to separate it from the previous ones
* why not annotating with NonNull like the others.